operations/service_web.go

```diff
@@ -134,12 +134,19 @@ func startWebService() cli.Command {
 	}
 }
 
-func gracefulShutdownForSIGTERM(ctx context.Context, servers []*http.Server, wait chan struct{}, catcher grip.Catcher) {
+func gracefulShutdownForSIGTERM(ctx context.Context, servers []*http.Server, wait chan struct{}, catcher grip.Catcher, env evergreen.Environment) {
 	defer recovery.LogStackTraceAndContinue("graceful shutdown")
 	sigChan := make(chan os.Signal, len(servers))
 	signal.Notify(sigChan, syscall.SIGTERM)
 
 	<-sigChan
+	// we got the signal, so modify the status endpoint and wait
+	err := env.SetShutdown()
+	if err != nil {
+		grip.Error(err)
+	}
+
+	time.Sleep(20 * time.Second)
```

Comment:
It's undesirable to sleep here. This will add 20 seconds per server when we deploy. I think just setting the shutdown flag is enough, if the load balancer doesn't check within that time then oh well

Reply:
Agreed on the specifics of what will happen, but I think @bsamek is more equipped to make judgement call than me.